libs/api/domains/regulations-admin/src/lib/client/regulationsAdmin.api.ts

```diff
@@ -0,0 +1,124 @@
+import { Inject } from '@nestjs/common'
+import { RESTDataSource, RequestOptions } from 'apollo-datasource-rest'
+import { DataSourceConfig } from 'apollo-datasource'
+import { EditDraftBody } from '../graphql/dto/editDraftRegulation.input'
+import { RegulationsAdminClientConfig } from '@island.is/clients/regulations-admin'
+import { ConfigType } from '@nestjs/config'
+import { DraftRegulationCancelModel } from '../graphql/models/draftRegulationCancel.model'
+import { DraftRegulationChangeModel } from '../graphql/models/draftRegulationChange.model'
+import {
+  CreateDraftRegulationCancelInput,
+  UpdateDraftRegulationCancelInput,
+  DeleteDraftRegulationCancelInput,
+  DeleteDraftRegulationChangeInput,
+  UpdateDraftRegulationChangeInput,
+  CreateDraftRegulationChangeInput,
+} from '../graphql/dto'
+import { DraftImpact } from '@island.is/regulations/admin'
+import { CreateDraftRegulationInput } from '../graphql/dto/createDraftRegulation.input'
+
+export class RegulationsAdminApi extends RESTDataSource {
```

Comment:
Yes absolutely. I have a change for that. I'm wondering since it's quite a change, and this has gone through thorough testing on our end as it is now, if it would be nice to get this to main and merge the client update afterwards?
As the API needed a bit of more change and refactor than I had expected initially.

Incoming changes WIP here: https://github.com/island-is/island.is/pull/10416/files